roadmap.md

```diff
@@ -0,0 +1,87 @@
+# Nix is everywhere and invisible
+
+Software developers use Nix as a matter of course every day, mostly without even noticing.
+Nix runs trivially, anywhere.
+
+For individuals to large organizations, Nix underpins the the entire software supply chain:
+- Developer tooling
+- Build automation
+- Binary distribution.
+
+To this end, the Nix team will work towards the following goals.
+
+## Make Nix easy to adopt
+
+  - Well-defined target user base
+  - Well-defined core user stories
+    - Ad hoc environments
+        - One-liner setup (nix-shell)
+    - Declarative environments
+        - One-liner setup ("templates")
+        - Easy modification/extension
+        - Easy and transparent usage ("direnv"/"lorri")
+    - Secret management as first-class citizen
```

Comment:
```suggestion
```

Not sure whether this is realistic in the roadmap time frame (except maybe via the ACL work).

Reply:
This is not a roadmap, this is a set of objectives. It does not have or imply a schedule. A roadmap is a sequence of sub-objectives to fulfill any may have a schedule.

Reply:
(eelco): Might be too low-level for the vision
(Robert): The ecosystem already has solutions for that, so probably not mandatory for Nix

Reply:
Hmm, is there any kind of a principle in the vision what should be allowed to evolve in the ecosystem and what ends up in the core?

Reply:
> The ecosystem already has solutions for that, so probably not mandatory for Nix

@robeth Which are those solutions? There was a brief lunch break debate at NixCon that e.g. a primitive in the Nix language   for dealing with secret values would be very helpful to enable more sophisticated solutions on top of that.

> is there any kind of a principle in the vision what should be allowed to evolve in the ecosystem and what ends up in the core?

@7c6f434c not yet, but I think we absolutely have to answer that question, somehow. For example, it could be implied by the vision statement, i.e. with everything that is not specifically defined by it. In any case we have to make explicit that we draw such a boundary, and make clear where that is.

Reply:
What currently says why pinning part of flakes should not be external add-on on equal footing with niv that people used for many of the same aims? I think that any «define basic logic and derive from there» kind of boundary will end up demanding a large redrawing of lines, so you probably need to define at least a _process_ for redrawing the boundary (which Nix currently lacks, judging from NixOS/RFCs#0049)

Reply:
> > The ecosystem already has solutions for that, so probably not mandatory for Nix
> 
> @roberth Which are those solutions? There was a brief lunch break debate at NixCon that e.g. a primitive in the Nix language for dealing with secret values would be very helpful to enable more sophisticated solutions on top of that.

Existing solutions: NixOps `deployment.keys`, sops-nix, agenix.
Potential solutions:
 - NixOS: make NixOps keys part of NixOS. For example `nixos-rebuild` could take secrets from `/root` and install them.
 - Nix: extend the string context to allow marking of secret values so that they can not be written to the store.
 - Nix: complicate the store concept. Not a fan of this.